addons/l10n_in_ewaybill_stock/models/l10n_in_ewaybill.py

```diff
@@ -0,0 +1,588 @@
+        copy=False,
+        tracking=True,
+        readonly=False
+    )
+
+    # Document number and date required in case of transportation mode is Rail, Air or Ship.
+    transportation_doc_no = fields.Char(
+        string="Transporter Doc No",
+        copy=False, tracking=True)
+    transportation_doc_date = fields.Date(
+        string="Transporter Doc Date",
+        copy=False,
+        tracking=True)
+
+    transporter_id = fields.Many2one("res.partner", "Transporter", copy=False, tracking=True)
+
+    error_message = fields.Html(readonly=True)
+    blocking_level = fields.Selection([
+        ("warning", "Warning"),
+        ("error", "Error")],
+        string="Blocking Level", readonly=True)
+
+    content = fields.Binary(compute='_compute_content', compute_sudo=True)
+    cancel_reason = fields.Selection(selection=[
+        ("1", "Duplicate"),
+        ("2", "Data Entry Mistake"),
+        ("3", "Order Cancelled"),
+        ("4", "Others"),
+    ], string="Cancel reason", copy=False, tracking=True)
+    cancel_remarks = fields.Char("Cancel remarks", copy=False, tracking=True)
+
+    def _compute_supply_type(self):
+        for ewaybill in self:
+            ewaybill.supply_type = ewaybill.picking_type_code == 'incoming' and 'I' or 'O'
+
+    @api.depends('picking_id')
+    def _compute_document_partners_details(self):
+        for ewaybill in self.filtered(lambda ewb: ewb.state == 'pending'):
+            picking_id = ewaybill.picking_id
+            if ewaybill.picking_type_code == 'incoming':
+                ewaybill.partner_bill_to_id = picking_id.company_id.partner_id
+                ewaybill.partner_bill_from_id = picking_id.partner_id
+                ewaybill.partner_ship_to_id = picking_id.picking_type_id.warehouse_id.partner_id
+                ewaybill.partner_ship_from_id = picking_id.partner_id
+            else:
+                ewaybill.partner_bill_to_id = picking_id.partner_id
+                ewaybill.partner_bill_from_id = picking_id.company_id.partner_id
+                ewaybill.partner_ship_to_id = picking_id.partner_id
+                ewaybill.partner_ship_from_id = picking_id.picking_type_id.warehouse_id.partner_id
+                if partner_invoice_id := ewaybill.picking_id._l10n_in_get_invoice_partner():
+                    ewaybill.partner_bill_to_id = partner_invoice_id
+
+            if (
+                ewaybill.picking_type_code == 'dropship' and
+                (dest_partner := ewaybill.picking_id._get_l10n_in_dropship_dest_partner())
+            ):
+                ewaybill.partner_ship_to_id = dest_partner
+                ewaybill.partner_ship_from_id = ewaybill.picking_id.partner_id
+
+    def _compute_is_fiscal_position_invisible(self):
+        for ewaybill in self:
+            fiscal_position = ewaybill.picking_id._l10n_in_get_fiscal_position()
+            ewaybill.is_fiscal_position_invisible = (
+                fiscal_position == self.env['account.fiscal.position'] or
```

Comment:
What is this supposed to do?  (when is it not True?)

Reply:
In case when sale/purchase order doesn't have a fiscal position it will return a empty record set since we want to get the fiscal from the order. But in case when there is no order then it will get `None/False`

Reply:
I guess returning a dict will be more meaning what we have already done `is_from_order`